mlflow/java/client/src/test/java/org/mlflow/tracking/HttpCallerTest.java

```diff
@@ -40,28 +44,32 @@ public MlflowHostCreds getHostCreds() {
     public void refresh() {
       // pass
     }
-  }, 4, 1, client);
+  }, 4, 1, 3, client);
 
   @BeforeSuite
   public void beforeAll() throws IOException {
     when(statusLine200.getStatusCode()).thenReturn(200);
-    when(statusLine429.getStatusCode()).thenReturn(429);
     when(response200.getStatusLine()).thenReturn(statusLine200);
-    when(entity200.getContent())
-            .thenReturn(
-                    new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8)))
-            .thenReturn(new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8)));
-    when(entity429.getContent())
-            .thenReturn(
-                    new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8)))
-            .thenReturn(new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8)));
+    when(entity200.getContent()).thenAnswer(
+      i -> new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8))
+    );
     when(response200.getEntity()).thenReturn(entity200);
+    when(statusLine429.getStatusCode()).thenReturn(429);
     when(response429.getStatusLine()).thenReturn(statusLine429);
+    when(entity429.getContent()).thenAnswer(
+      i -> new ByteArrayInputStream(expectedResponseText.getBytes(StandardCharsets.UTF_8))
+    );
```

Comment:
Use `thenAnswer` instead of `thenReturn` to ensure that a new `ByteArrayInputStream` is constructed each time there's an interaction with the mock (otherwise, the stream content will only available to the first caller of the mock who consumes it). For more context, see https://stacktraceguru.com/unittest/thenreturn-vs-thenanswer